matRad_rayTracing.m

```diff
@@ -0,0 +1,156 @@
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% matRad visualization of two-dimensional dose distributions on ct including
+% segmentation
+% 
+% call
+%   [radDepthCube,geoDistCube] = matRad_rayTracing(stf,ct,V,lateralCutoff)
+%
+% input
+%   stf:           matRad steering information struct of one beam
+%   ct:            ct cube
+%   V:             linear voxel indices e.g. of voxels inside patient.
+%   lateralCutoff: lateral cut off used for ray tracing
+
+%
+% output
+%   radDepthCube:  radiological depth cube in the ct.cube dimensions
+%   geoDistCube:   geometrical distance cube in the ct.cube dimensions
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% Copyright 2015, Mark Bangert, on behalf of the matRad development team
+%
+% m.bangert@dkfz.de
+%
+% This file is part of matRad.
+%
+% matrad is free software: you can redistribute it and/or modify it under
+% the terms of the GNU General Public License as published by the Free
+% Software Foundation, either version 3 of the License, or (at your option)
+% any later version.
+%
+% matRad is distributed in the hope that it will be useful, but WITHOUT ANY
+% WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+% FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+% details.
+%
+% You should have received a copy of the GNU General Public License in the
+% file license.txt along with matRad. If not, see
+% <http://www.gnu.org/licenses/>.
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% set up rad depth cube for results
+radDepthCube = inf*ones(size(ct.cube));
+
+% set up coordinates of all voxels in cube
+[yCoords_vox, xCoords_vox, zCoords_vox] = ind2sub(size(ct.cube),1:numel(ct.cube));
+
+xCoords = xCoords_vox(:)*ct.resolution.x-stf.isoCenter(1);
+yCoords = yCoords_vox(:)*ct.resolution.y-stf.isoCenter(2);
+zCoords = zCoords_vox(:)*ct.resolution.z-stf.isoCenter(3);
+
+% Rotation around Z axis (gantry)
+inv_rotMx_XY_T = [ cosd(-stf.gantryAngle) sind(-stf.gantryAngle) 0;
+                  -sind(-stf.gantryAngle) cosd(-stf.gantryAngle) 0;
+                                        0                      0 1];
+
+% Rotation around Y axis (Couch movement)
+inv_rotMx_XZ_T = [cosd(-stf.couchAngle) 0 -sind(-stf.couchAngle);
+                                      0 1                      0;
+                  sind(-stf.couchAngle) 0  cosd(-stf.couchAngle)];
+ 
+
+coords_bev = [xCoords yCoords zCoords]*inv_rotMx_XZ_T*inv_rotMx_XY_T;             
+              
+% set up ray matrix direct behind last voxel
+rayMx_bev_z = max(coords_bev(V,2));
+
+
+xCoords = xCoords-stf.sourcePoint(1);
+yCoords = yCoords-stf.sourcePoint(2);
+zCoords = zCoords-stf.sourcePoint(3);
+coords  = [xCoords yCoords zCoords];
+    
+% calculate geometric distances
+geoDistCube = sqrt(sum(coords.^2,2));
```

Comment:
can be checked if really needed! not needed for calcParticleDose iff SSD calculation moved to genSTF function